src/WireMock.Net/Server/WireMockServer.Admin.cs

```diff
@@ -600,6 +601,26 @@ private IResponseMessage RequestsFind(IRequestMessage requestMessage)
 
         return ToJson(result);
     }
+
+    private IResponseMessage RequestFindByMappingGuid(IRequestMessage requestMessage)
+    {
+        if (requestMessage.Query != null &&
+            requestMessage.Query.TryGetValue("mappingGuid", out var value) &&
+            Guid.TryParse(value.ToString(), out var mappingGuid)
+        )
+        {
+            var logEntry = LogEntries.SingleOrDefault(le => !le.RequestMessage.Path.StartsWith("/__admin/") && le.MappingGuid == mappingGuid);

```

Comment:
There could be more than one matching request, and in this case `SingleOrDefault` will throw.